app/Models/OAuth/Token.php

```diff
@@ -43,6 +60,38 @@ public function revoke()
         return $saved;
     }
 
+    public function scopeValidAt($query, $time)
+    {
+        return $query->where('revoked', false)->where('expires_at', '>', $time);
+    }
+
+    public function validate()
+    {
+        if (empty($this->scopes)) {
+            throw new MissingScopeException([], 'Tokens without scopes are not valid.');
+        }
+
+        if ($this->isClientCredentials() && in_array('*', $this->scopes, true)) {
+            throw new MissingScopeException(['*'], '* is not allowed with Client Credentials');
+        }
+
+        if (in_array('bot', $this->scopes, true)) {
+            if (count($this->scopes) !== 1) {
+                throw new MissingScopeException(['bot'], 'bot scope is not allowed with other scopes.');
```

Comment:
`bot` allows grants using `client_credentials` to have a user assigned to the `auth()` guard (otherwise, none of the things relying on `auth()->user()` would work) which is incompatible with how `client_credentials` grants currently work, where everything it uses an anonymous user.

`bot` is a special scope that
- is only available to OAuth clients owned by users in the chat bot group.
- is only allowed with `client_credentials`.
- lets the request have an auth'd user.

the alternative is some combination scope like `chat.write identify`, but it's still only going to be available to bots only.

There's also going to be the issue when using `client_credentials`, the `client_secret` is going to have to be deployed to whatever environment is going to be requesting the token; that passport's scopes aren't restricted by client and any arbitrary scope can be requested is an issue - this is one of the reasons `client_credentials` grants are currently forced to be guest only.